### ISSUE TITLE
New switch between ADV/NVL mode.

### ISSUE THREAD

**Issue by irlPM** [07th-mod/higurashi-sui-merger#1]
**I found that there is indeed the command to NVL and ADV in the sui_script, so I thought about using this tool to pull out and copy to our script, basically the same way  the OST was implemented.**

- **_About the Sui commands_**
I saw different episodes in youtube and used the OCR to match the line in sui script and pretty much the conclusion is that this command draws the NVL mode `<ins type="TEXTBOX_COMMAND" textbox_disable="1" align="2" ></ins>`   <ins type="TEXTBOX_COMMAND" textbox_disable="0" align="0" ></ins> and `<ins type="TEXTBOX_COMMAND" textbox_disable="0" align="0" ></ins>` draws ADV mode.

- _**Our Script**_
It's pretty simple,  NVL mode `ModEnableNVLModeInADVMode()` and to draw the ADV mode again, `ModDisableNVLModeInADVMode()`

- _**Conclusion**_
 I think the best way to handle this is the tool to work the same way as the ost, find the nearest japanese line to the sui command, ''translate'' and copy to our script in the same japanese line. @enumag advised me to use the `DisableWindow()` command before a `ModEnableNVLModeInADVMode()` line, so that would have to be considered to.

.

**Comment by enumag** [comment]
Note: `DisableWindow()` should already be in place where needed. Just add the `ModEnableNVLModeInADVMode()` / `ModDisableNVLModeInADVMode()` after that.

**Comment by drojf** [comment]
I think all you have to do is re-run the merger on the latest version of your script that you want to patch in. Assuming that works, you will then have json-encoded commands in your script as comments. 

At this point, you can do whatever you want using your own tools. You can use any JSON library to import the commands, and look for the textbox_disable. 

I can do the second step using my python script too, but I assumed that you guys would want to use your own tools to do it, otherwise there will be a back-and-forth if the results aren't as you want. And I don't have a proper setup to test the game (I haven't actually played the game lol).

So in short, give me the latest version of the script you want to insert into, and I'll insert the JSON into the script (or you guys can do it if you know how). Then you can process it how you like.

**Comment by drojf** [comment]
I should expand that this is how I expected the workflow to go

XML/JSON Insertion:

0. download a copy of my script from github
1. put the xml file in the 'data' folder
2. put the scripts you want to merge the xml into in the 'input' folder
3. run 'main.py'
4. output is the files which have json embedded in them as comments

Then, using your own tool

0. Iterate through the script line by line
1. If you encounter a JSON comment, Use a JSON library to extract the xml commands (which are now comments in your scripts)
2. removing the JSON command, and insert your own commands as applicable.

My own workflow is the same - the file 'useInformation.py' performs the "using your own tool" steps:

```
       if '//JSON_INSTRUCTIONS' in line:
            dialogue_num = None #unique reference for this block of instructions
            line_json = line.replace('//JSON_INSTRUCTIONS:','')
            data = json.loads(line_json)
            output_instructions = []
            for instruction in data:
                if instruction['type'] == 'BGM_PLAY': 'bgm_file': 'HM01_05', 'volume': '188', 'single_play': '0'}
                    # do some stuff here!
```

**Comment by enumag** [comment]
@drojf Nice! I should be able to take care of that. Connecting my compiler with your merger might be useful in the future as well.

**Comment by irlPM** [comment]
As @drojf  asked here is all the scripts until the last released episode.
[All Scripts.zip](https://github.com/07th-mod/higurashi-sui-merger/files/2019913/All.Scripts.zip)

**Comment by drojf** [comment]
I think the ADV/NVL switching is fully implemented now, so I'm going to close this issue.

Also, since there is a newer C# version of the merger (instead of this python version), please use that one instead: https://github.com/07th-mod/cs-sui-merger

I will archive this (the python version) repo after closing this issue.
